wp-admin/js/widgets/media-widgets.js

```diff
@@ -121,8 +121,34 @@ wp.mediaWidgets = ( function( $ ) {
 							embedLinkView.dfd.abort();
 						}
 
+						fetchSuccess = function( response ) {
+							embedLinkView.renderoEmbed({
+								data: {
+									body: response
+								}
+							});
+
+							$( '#embed-url-field' ).removeClass( 'invalid' );
+							embedLinkView.views.parent.views.parent.views.get( '.media-frame-toolbar' )[0].$el.find( '.media-button-select' ).prop( 'disabled', false );
+						};
+
+						urlParser = document.createElement( 'a' );
+						urlParser.href = embedLinkView.model.get( 'url' );
+						matches = urlParser.pathname.toLowerCase().match( /\.(\w+)$/ );
```

Comment:
Yeah, it's a valid concern. Nevertheless, file extensions are relied on in the video shortcode for rendering: https://github.com/WordPress/wordpress-develop/blob/c449b34f6c35e179925110e076797a9c7df4a7f2/src/wp-includes/media.php#L2494-L2498

So I don't think we're doing anything that isn't aligned with what core is already doing.